packages/asset-swapper/src/utils/market_operation_utils/index.ts

```diff
@@ -120,6 +122,11 @@ export class MarketOperationUtils {
             [orderFillableAmounts, liquidityProviderAddress, ethToMakerAssetRate, dexQuotes],
             rfqtIndicativeQuotes,
         ] = await Promise.all([samplerPromise, rfqtPromise]);
+
+        if (_opts.quoteReporter) {
```

Comment:
What do you think about null object pattern here. The defaults would default to a `new NullQuoteReporter()` which has `trackDexSamples` etc but just does nothing.

That way we can default it in params and not have to do an existence check all the time

Reply:
If we only had to define 1 fn on `NullQuoteReporter` I'd be down, but at this point there are 6 public functions which makes this feel a bit janky to me